base/src/pos.rs

```diff
@@ -135,64 +135,62 @@ pub struct Span {
 }
 
 impl Span {
-    pub fn containment(&self, location: &Location) -> Ordering {
+    pub fn contains(self, other: Span) -> bool {
+        self.start <= other.start && other.end <= self.end
+    }
+
+    pub fn containment(self, location: &Location) -> Ordering {
         use std::cmp::Ordering::*;
+
         match (location.cmp(&self.start), location.cmp(&self.end)) {
             (Equal, _) | (Greater, Less) => Equal,
             (Less, _) => Less,
             (_, Equal) | (_, Greater) => Greater,
         }
     }
 
-    pub fn containment_exclusive(&self, location: &Location) -> Ordering {
+    pub fn containment_exclusive(self, location: &Location) -> Ordering {
         if self.end == *location {
             Ordering::Greater
         } else {
             self.containment(location)
         }
     }
-}
 
-#[derive(Copy, Clone, PartialEq, Debug)]
-pub struct Spanned<T> {
-    pub span: Span,
-    pub value: T,
-}
-
-impl<T: fmt::Display> fmt::Display for Spanned<T> {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "{}: {}", self.span.start, self.value)
+    pub fn merge(self, other: Span) -> Option<Span> {
```

Comment:
I was thinking that this would allow us to expand the spans during parsing, but we are only emitting located tokens right now....